docs/source/reservations/reservations.rst

```diff
@@ -170,32 +171,141 @@ In case you want to specify the DWS configuration for each job/cluster, you can
+
+* Flex-start with queued provisioning, where GKE allocates all requested resources at the same time.
+
+  * To use this configuration, you have to add the ``flex-start`` and ``enable-queued-provisioning`` flags when you create the node pool.
+
+  * Flex-start with queued provisioning configuration is recommended for the medium to large workload, which means that the workload can run on multiple nodes. Your workload requires multiple resources and can't start running until all nodes are provisioned and ready at the same time. For example, this configuration works well if you are running distributed machine learning training workloads.
+
+See `GKE DWS documentation <https://cloud.google.com/kubernetes-engine/docs/concepts/dws>`_ for more details.
+
+.. note::
+
+    If you are using Kueue, and ``kubernetes.kueue.local_queue_name`` is specified, SkyPilot will automatically enable the flex-start with queued provisioning mode, otherwise it will use the flex-start mode.
+
+
+Flex-start
+^^^^^^^^^^
+
+To launch clusters or managed jobs with flex-start configuration:
 
-DWS is also supported on Google Kubernetes Engine (GKE) with Kueue. To enable DWS on GKE, you need to set up your GKE cluster with Kueue and DWS; see the `GCP official docs <https://cloud.google.com/kubernetes-engine/docs/how-to/provisioningrequest>`__.
+1. Follow the `official documentation <https://cloud.google.com/kubernetes-engine/docs/how-to/dws-flex-start-training#node-pool-flex>`_ to create a node pool with flex-start enabled.
 
-To launch a SkyPilot cluster or job on GKE with DWS, you can specify the DWS configuration in the SkyPilot task YAML:
+2. Configure the following fields in ``~/.sky/config.yaml``:
 
 .. code-block:: yaml
 
+    kubernetes:
+      # provision_timeout: 1200
+      autoscaler: gke
+      dws:
+        enabled: true
+
+When flex-start is enabled, the default ``provision_timeout`` is set to ``600`` seconds (10 minutes). If you encounter provisioning timeout issues, you can increase this value in your configuration. For example, set it to ``1200`` seconds (20 minutes) or higher depending on your workload and cluster size.
+
+3. Launch your clusters or managed jobs.
+
+.. code-block:: yaml
+
+    name: flex-start
+
+    resources:
+      infra: k8s
+      accelerators: L4:1
+
+    num_nodes: 1
+
+In case you want to specify the DWS configuration for each job/cluster, you can also specify the configuration in the SkyPilot task YAML (see :ref:`here <config-client-job-task-yaml>`):
+
+.. code-block:: yaml
+
+    name: flex-start
+
+    resources:
+      infra: k8s
+      accelerators: L4:1
+
+    num_nodes: 1
+
     config:
       kubernetes:
-        pod_config:
-          metadata:
-            annotations:
-              provreq.kueue.x-k8s.io/maxRunDurationSeconds: "3600"
-        provision_timeout: 900
+        dws:
+          enabled: true
+
+Flex-start with queued provisioning
```

Comment:
The docs should be changed, i.e. we should not separate the case by `Flex-start with queued provisioning` but using `Kueue` or not.